test/hexpm/short_urls/short_url_test.exs

```diff
@@ -0,0 +1,37 @@
+defmodule Hexpm.ShortURLs.ShortURLTest do
+  use Hexpm.DataCase, async: true
+  alias Hexpm.ShortURLs.ShortURL
+
+  describe "changeset/1" do
+    test "with correct params, creates a new short url" do
+      params = %{"url" => "https://diff.hex.pm?diff[]=ecto:3.0.1:3.0.4"}
+      %{valid?: valid?, changes: changes} = ShortURL.changeset(params)
+      assert valid? == true
+      assert String.length(changes.short_code) == 5
+    end
+
+    test "valid when redirecting to hex.pm" do
+      params = %{"url" => "https://hex.pm"}
+      %{valid?: valid?} = ShortURL.changeset(params)
+      assert valid? == true
+    end
+
+    test "valid when redirecting to a complex subdomain on hex.pm" do
+      params = %{"url" => "https://www.links.hex.pm"}
+      %{valid?: valid?} = ShortURL.changeset(params)
+      assert valid? == true
+    end
```

Comment:
we have two different styles in this describe: pattern matching on changeset, or assigning changeset to the variable. How about we standardise on the latter?

```suggestion
    test "with correct params, creates a new short url" do
      params = %{"url" => "https://diff.hex.pm?diff[]=ecto:3.0.1:3.0.4"}
      changeset = ShortURL.changeset(params)
      assert changeset.valid?
      assert String.length(changeset.changes.short_code) == 5
    end

    test "valid when redirecting to hex.pm" do
      params = %{"url" => "https://hex.pm"}
      changeset = ShortURL.changeset(params)
      assert changeset.valid?
    end

    test "valid when redirecting to a complex subdomain on hex.pm" do
      params = %{"url" => "https://www.links.hex.pm"}
      changeset = ShortURL.changeset(params)
      assert changeset.valid?
    end
```